packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts

```diff
@@ -136,6 +141,55 @@ export class TestRunnerService {
 		return mainConnectionData?.[0]?.json ?? {};
 	}
 
+	private extractMetricsFromEvaluationResult(result: IDataObject, metrics: Set<string>) {
+		const extractedMetrics: Record<string, number> = {};
+
+		for (const metric of metrics) {
+			if (typeof result[metric] === 'number') {
+				extractedMetrics[metric] = result[metric];
+			}
```

Comment:
This decision was made to trim down the scope of the first iteration.
Providing feedback is not trivial here, because the evaluation is running asynchronously. But I agree we need some way to collect errors or warning during the test run, we will discuss it.